lmdeploy/turbomind/turbomind.py

```diff
@@ -238,53 +239,35 @@ def _from_hf(self, model_source: ModelSource, model_path: str,
             'Plz try pytorch engine instead.')
 
         # convert transformers model into turbomind model format
-        model_arch, _ = get_model_arch(model_path)
-        data_type = 'fp16'
-        output_format = 'fp16'
-        inferred_model_format = get_model_format(SUPPORTED_ARCHS[model_arch],
-                                                 engine_config.model_format)
-        cfg = TurbomindModelConfig.from_engine_config(engine_config)
         match_name = best_match_model(model_path)
-        # for session len
-        cfg.model_name = match_name \
-            if match_name is not None else 'base'
-        if inferred_model_format.find('awq') != -1:
-            cfg.weight_type = 'int4'
-            output_format = 'w4'
-            data_type = 'int4'
-            cfg.group_size = 128
-            if inferred_model_format == 'xcomposer2-awq':
-                output_format = 'plora-w4'
-        else:
-            output_format = update_output_format(cfg.model_name,
-                                                 inferred_model_format,
-                                                 model_path, output_format)
-            data_type = output_format
-            update_config_weight_type(output_format, cfg)
-            if inferred_model_format == 'xcomposer2':
-                output_format = 'plora'
-
-        input_model = INPUT_MODELS.get(inferred_model_format)(
+        input_model_name = get_input_model_registered_name(
+            model_path, engine_config.model_format)
+        input_model = INPUT_MODELS.get(input_model_name)(
             model_path=model_path, tokenizer_path=model_path, ckpt_path=None)
 
-        output_model = OUTPUT_MODELS.get(output_format)(
-            input_model=input_model, cfg=cfg, to_file=False, out_dir='')
-
-        cfg = output_model.cfg
-        if engine_config.session_len is not None:
-            cfg.session_len = engine_config.session_len
-
-        self.model_name = cfg.model_name
-        self.config = cfg
-        self.data_type = data_type
-
-        logger.warning(f'model_config:\n\n{cfg.toini()}')
+        output_model_name, cfg = get_output_model_registered_name_and_config(
+            model_path=model_path,
+            model_format=engine_config.model_format,
+            group_size=0)
+        self.config = TurbomindModelConfig.from_engine_config(engine_config)
+        self.config.update(cfg)
```

Comment:
If the user doesn't set `session_len` in backend_config, the `num_tokens_per_iter` and `max_prefill_iters` will not set in `TurbomindModelConfig.from_engine_config`.

May move [this code](https://github.com/InternLM/lmdeploy/blob/42267a8f3bf11c2579dd073e30bee95bd0149cba/lmdeploy/turbomind/deploy/target_model/base.py#L100-L106) to `update` function.

Reply:
It seems hasn't been fixed yet.

If the user doesn't set session_len when init the pipeline, the value of `num_tokens_per_iter` and `max_prefill_iters` are not right.

Reply:
Updated. It should be fixed now